sdks/java/extensions/avro/src/main/java/org/apache/beam/sdk/extensions/avro/io/SerializableAvroCodecFactory.java

```diff
@@ -50,6 +50,9 @@ class SerializableAvroCodecFactory implements Externalizable {
   private static final Pattern deflatePattern = Pattern.compile(DEFLATE_CODEC + "-(?<level>-?\\d)");
   private static final Pattern xzPattern = Pattern.compile(XZ_CODEC + "-(?<level>\\d)");
 
+  // Don't reference `DataFileConstants.ZSTANDARD_CODEC` directly for Avro 1.8 compat
+  private static final Pattern zstdPattern = Pattern.compile("zstandard\\[(?<level>\\d+)\\]");
```

Comment:
Note: DataFileConstants.STANDARD_CODEC was added in [Avro 1.9](https://github.com/apache/avro/blob/release-1.9.2/lang/java/avro/src/main/java/org/apache/avro/file/DataFileConstants.java#L41C30-L41C45), so I didn't want to reference it here and break 1.8 users. When Avro 1.8 is dropped it can be used directly

Reply:
Sorry for late question. zstandard compression level range from negative 7 to 22. This regex does not handle negative, is this a concern?

Reply:
oh great catch! added a `-?` to the regex and updated the tests.